### ISSUE TITLE
Question about using automatic differentiation of deepxde for PDE

### ISSUE THREAD

**Issue by KenzhebekYerzhan** [lululxvi/deepxde#377]
Hi dear LuLu, I hope you are very well.

I would like to ask you for advice, I am trying to solve the following PDE:

![Screenshot 2021-09-18 145239](https://user-images.githubusercontent.com/61427499/133883052-848b9b44-690f-4e0d-8d99-83d402475040.jpg)

I would like to know how to use automatic differentiation (dde.grad.jacobian/hessian) for this situation. I know that I could have used dde.grad.hessian to calculate the second order derivatives if it weren't for  _k_. The _k_ is function depending on _x_ .

Thanks a lot in advance for this wonderful library. 
Best,
Yerzhan

**Comment by smurf-17** [comment]
I guess you can try like 
dp  = tf.gradients(p,x)[0]
p_x = dp[:,0:1]
func_k = p_x*k
dfunc_k= tf.gradients(func_k,x)
func_k_x =dfunc_k[:,0:1]

Just check it once Iam not 100% sure if this is correct

**Comment by lululxvi** [comment]
```python
dpdx = dde.grad.jacobian(...)
d2 = dde.grad.jacobian(k * dpdx, x, ...)
```

**Comment by KenzhebekYerzhan** [comment]
> ```python
> dpdx = dde.grad.jacobian(...)
> d2 = dde.grad.jacobian(k * dpdx, x, ...)
> ```

Thanks dear @lululxvi, but actually  I am trying to solve the following PDE:
![image](https://user-images.githubusercontent.com/61427499/135444715-2dec1e9c-01d2-40be-aaf0-c22e0ad1712a.png)

And I'm not sure if the same output tensor y can be used to compute the gradient P and S in the function def pde()?
Will it be correct if I write the code like this?

def pde(x, y):
ds_t = dde.grad.jacobian(y, x, i=0, j=1)
dpdx = dde.grad.jacobian(y, x, i=0, j=0) //y->p?
d2 = dde.grad.jacobian(k*dpdx, x, i=0, j=0)
return ds_t + d2

or

def pde(x, s, p):
ds_t = dde.grad.jacobian(s, x, i=0, j=1)
dpdx = dde.grad.jacobian(p, x, i=0, j=0) //y->p?
d2 = dde.grad.jacobian(k*dpdx, x, i=0, j=0)
return ds_t + d2

how can I solve in this situation where I should have 2 output? Thanks.

**Comment by lululxvi** [comment]
y = [S, P], or [P, S]. You make the choice.

The k part is correct.

**Comment by KenzhebekYerzhan** [comment]
> y = [S, P], or [P, S]. You make the choice.
> 
> The k part is correct.

Thanks Professor @lululxvi for your response. So, in my case if i choose y = [P,S], is PDE will true? :
def pde(x, y):
  ds_t = dde.grad.jacobian(y, x, i=1, j=1) 
  (Here, i = 1 ->means second output S?,  j = 1 -> means compute gradient respect to second input t?)
  dpdx = dde.grad.jacobian(y, x, i=0, j=0) 
  d2 = dde.grad.jacobian(k*dpdx, x, i=0, j=0) (Here our k depends on S)
  return ds_t + d2
  
bcL_P = dde.DirichletBC(geomtime, funcL_P, boundary_l, component=0)
bcr_P = dde.DirichletBC(geomtime, funcR_P, boundary_r, component=0)
bcL_S = dde.DirichletBC(geomtime, funcBL_S, boundary_l, component=1)
bcr_S = dde.DirichletBC(geomtime, funcBR_S, boundary_r, component=1)

ic_P = dde.IC(geomtime, funcI_P, lambda _, on_initial: on_initial, component=0)
ic_S = dde.IC(geomtime, funcI_S, lambda _, on_initial: on_initial, component=1)
data = dde.data.TimePDE(
    geomtime,
    pde,
    [bcL_P, bcr_P, bcL_S, bcr_S, ic_P, ic_S],
    num_domain=2540,
    num_boundary=80,
    num_initial=160,
    num_test=100
)

layer_size = [2] + [50] * 3 + [2]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.maps.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)
model.compile("adam", lr=0.001)
losshistory, train_state = model.train(epochs=10000)

In our case: S depends on x and t, k depends on S
S component doesn't change in the Plot on t (time)
